src/map/clif.c

```diff
@@ -4032,51 +4032,58 @@ static void clif_misceffect(struct block_list *bl, int type)
 /// 0229 <id>.L <body state>.W <health state>.W <effect state>.L <pk mode>.B (ZC_STATE_CHANGE3)
 static void clif_changeoption(struct block_list *bl)
 {
-	unsigned char buf[32];
-	struct status_change *sc;
-	struct map_session_data* sd;
-
 	nullpo_retv(bl);
 
-	if ( !(sc = status->get_sc(bl)) && bl->type != BL_NPC ) return; //How can an option change if there's no sc?
+	struct map_session_data *sd = BL_CAST(BL_PC, bl);
```

Comment:
You only need `sd` if `if (sc == NULL && bl->type != BL_NPC)` is `false`, so the declaration should be moved below this check.